sorts/cyclic_sort.py

```diff
@@ -0,0 +1,52 @@
+"""
+This is a pure Python implementation of the Cyclic Sort algorithm.
+
+For doctests run following command:
+python -m doctest -v cyclic_sort.py
+or
+python3 -m doctest -v cyclic_sort.py
+For manual testing run:
+python cyclic_sort.py
+"""
+
+
+def cyclic_sort(nums: list) -> list:
+    """
+    Sorts the input list in-place using the Cyclic Sort algorithm.
+
+    :param nums: List of integers to be sorted.
+    :return: The same list sorted in ascending order.
+
+    Time complexity: O(n), where n is the number of elements in the list.
+
+    Examples:
```

Comment:
Similar to above, I'd recommend adding an example to demonstrate what happens if the list contains elements outside the expected range (a non-positive integer) or duplicates are present in the list. This would show why this algorithm should not be used for those sets of values.